### ISSUE TITLE
[6.2] Undefined array key in command translation:extract

### ISSUE THREAD

**Issue by knallcharge** [symfony/symfony#48419]
### Symfony version(s) affected

6.2.0

### Description

After updating to Symfony 6.2.0 (from 6.1.8) the command "translation:extract" throws the following error:

```
php bin/console translation:extract --force de

Translation Messages Extractor and Dumper
=========================================

 // Generating "de" translation files for "default directory"

 // Parsing templates...

11:08:16 ERROR     [main] ErrorException: Warning: Undefined array key 2 (uncaught error) at (...)\vendor\symfony\translation\Extractor\Visitor\AbstractVisitor.php line 56 while running console command `translation:extract` ["error" => ErrorException { …}]
11:08:16 WARNING   [main] Command `translation:extract` exited with status code 1

In AbstractVisitor.php line 56:

  Warning: Undefined array key 2
```

### How to reproduce

run the command
php bin/console translation:extract --force de

or
php bin/console translation:extract --dump-messages de

**Comment by stof** [comment]
Can you run the command in verbose mode so that the error rendering includes the stack trace ?

**Comment by knallcharge** [comment]
Sure, here you go:

```
php bin/console translation:extract -vvv --force de

Translation Messages Extractor and Dumper
=========================================

 // Generating "de" translation files for "default directory"

 // Parsing templates...

12:33:19 ERROR     [main] ErrorException: Warning: Undefined array key 2 (uncaught error) at (...)\vendor\symfony\translation\Extractor\Visitor\AbstractVisitor.php line 56 while running console command `translation:extract`
[
  "error" => ErrorException {
    #message: "Warning: Undefined array key 2"
    #code: 0
    #file: "(...)\vendor\symfony\translation\Extractor\Visitor\AbstractVisitor.php"
    #line: 56
    #severity: E_WARNING
    trace: {
      (...)\vendor\symfony\translation\Extractor\Visitor\AbstractVisitor.php:56 { …}
      (...)\vendor\symfony\translation\Extractor\Visitor\TransMethodVisitor.php:45 { …}
      (...)\vendor\nikic\php-parser\lib\PhpParser\NodeTraverser.php:123 { …}
      (...)\vendor\nikic\php-parser\lib\PhpParser\NodeTraverser.php:223 { …}
      (...)\vendor\nikic\php-parser\lib\PhpParser\NodeTraverser.php:114 { …}
      (...)\vendor\nikic\php-parser\lib\PhpParser\NodeTraverser.php:146 { …}
      (...)\vendor\nikic\php-parser\lib\PhpParser\NodeTraverser.php:223 { …}
      (...)\vendor\nikic\php-parser\lib\PhpParser\NodeTraverser.php:114 { …}
      (...)\vendor\nikic\php-parser\lib\PhpParser\NodeTraverser.php:223 { …}
      (...)\vendor\nikic\php-parser\lib\PhpParser\NodeTraverser.php:114 { …}
      (...)\vendor\nikic\php-parser\lib\PhpParser\NodeTraverser.php:146 { …}
      (...)\vendor\nikic\php-parser\lib\PhpParser\NodeTraverser.php:223 { …}
      (...)\vendor\nikic\php-parser\lib\PhpParser\NodeTraverser.php:114 { …}
      (...)\vendor\nikic\php-parser\lib\PhpParser\NodeTraverser.php:223 { …}
      (...)\vendor\nikic\php-parser\lib\PhpParser\NodeTraverser.php:114 { …}
      (...)\vendor\nikic\php-parser\lib\PhpParser\NodeTraverser.php:223 { …}
      (...)\vendor\nikic\php-parser\lib\PhpParser\NodeTraverser.php:114 { …}
      (...)\vendor\nikic\php-parser\lib\PhpParser\NodeTraverser.php:223 { …}
      (...)\vendor\nikic\php-parser\lib\PhpParser\NodeTraverser.php:91 { …}
      (...)\vendor\symfony\translation\Extractor\PhpAstExtractor.php:56 { …}
      (...)\vendor\symfony\translation\Extractor\ChainExtractor.php:48 { …}
      (...)\vendor\symfony\framework-bundle\Command\TranslationUpdateCommand.php:395 { …}
      (...)\vendor\symfony\framework-bundle\Command\TranslationUpdateCommand.php:200 { …}
      (...)\vendor\symfony\console\Command\Command.php:312 { …}
      (...)\vendor\symfony\console\Application.php:1038 { …}
      (...)\vendor\symfony\framework-bundle\Console\Application.php:88 { …}
      (...)\vendor\symfony\console\Application.php:312 { …}
      (...)\vendor\symfony\framework-bundle\Console\Application.php:77 { …}
      (...)\vendor\symfony\console\Application.php:168 { …}
      (...)\vendor\symfony\runtime\Runner\Symfony\ConsoleApplicationRunner.php:54 { …}
      (...)\vendor\autoload_runtime.php:29 { …}
      (...)\bin\console:11 {
        ›
        › require_once dirname(__DIR__) . '/vendor/autoload_runtime.php';
        ›
        arguments: {
          "(...)\vendor\autoload_runtime.php"
        }
      }
    }
  }
]
12:33:19 WARNING   [main] Command `translation:extract` exited with status code 1

In AbstractVisitor.php line 56:

  [ErrorException]
  Warning: Undefined array key 2


Exception trace:
  at (...)\vendor\symfony\translation\Extractor\Visitor\AbstractVisitor.php:56
 Symfony\Component\Translation\Extractor\Visitor\AbstractVisitor->getStringArguments() at (...)\vendor\symfony\translation\Extractor\Visitor\TransMethodVisitor.php:45
 Symfony\Component\Translation\Extractor\Visitor\TransMethodVisitor->enterNode() at (...)\vendor\nikic\php-parser\lib\PhpParser\NodeTraverser.php:123
 PhpParser\NodeTraverser->traverseNode() at (...)\vendor\nikic\php-parser\lib\PhpParser\NodeTraverser.php:223
 PhpParser\NodeTraverser->traverseArray() at (...)\vendor\nikic\php-parser\lib\PhpParser\NodeTraverser.php:114
 PhpParser\NodeTraverser->traverseNode() at (...)\vendor\nikic\php-parser\lib\PhpParser\NodeTraverser.php:146
 PhpParser\NodeTraverser->traverseNode() at (...)\vendor\nikic\php-parser\lib\PhpParser\NodeTraverser.php:223
 PhpParser\NodeTraverser->traverseArray() at (...)\vendor\nikic\php-parser\lib\PhpParser\NodeTraverser.php:114
 PhpParser\NodeTraverser->traverseNode() at (...)\vendor\nikic\php-parser\lib\PhpParser\NodeTraverser.php:223
 PhpParser\NodeTraverser->traverseArray() at (...)\vendor\nikic\php-parser\lib\PhpParser\NodeTraverser.php:114
 PhpParser\NodeTraverser->traverseNode() at (...)\vendor\nikic\php-parser\lib\PhpParser\NodeTraverser.php:146
 PhpParser\NodeTraverser->traverseNode() at (...)\vendor\nikic\php-parser\lib\PhpParser\NodeTraverser.php:223
 PhpParser\NodeTraverser->traverseArray() at (...)\vendor\nikic\php-parser\lib\PhpParser\NodeTraverser.php:114
 PhpParser\NodeTraverser->traverseNode() at (...)\vendor\nikic\php-parser\lib\PhpParser\NodeTraverser.php:223
 PhpParser\NodeTraverser->traverseArray() at (...)\vendor\nikic\php-parser\lib\PhpParser\NodeTraverser.php:114
 PhpParser\NodeTraverser->traverseNode() at (...)\vendor\nikic\php-parser\lib\PhpParser\NodeTraverser.php:223
 PhpParser\NodeTraverser->traverseArray() at (...)\vendor\nikic\php-parser\lib\PhpParser\NodeTraverser.php:114
 PhpParser\NodeTraverser->traverseNode() at (...)\vendor\nikic\php-parser\lib\PhpParser\NodeTraverser.php:223
 PhpParser\NodeTraverser->traverseArray() at (...)\vendor\nikic\php-parser\lib\PhpParser\NodeTraverser.php:91
 PhpParser\NodeTraverser->traverse() at (...)\vendor\symfony\translation\Extractor\PhpAstExtractor.php:56
 Symfony\Component\Translation\Extractor\PhpAstExtractor->extract() at (...)\vendor\symfony\translation\Extractor\ChainExtractor.php:48
 Symfony\Component\Translation\Extractor\ChainExtractor->extract() at (...)\vendor\symfony\framework-bundle\Command\TranslationUpdateCommand.php:395
 Symfony\Bundle\FrameworkBundle\Command\TranslationUpdateCommand->extractMessages() at (...)\vendor\symfony\framework-bundle\Command\TranslationUpdateCommand.php:200
 Symfony\Bundle\FrameworkBundle\Command\TranslationUpdateCommand->execute() at (...)\vendor\symfony\console\Command\Command.php:312
 Symfony\Component\Console\Command\Command->run() at (...)\vendor\symfony\console\Application.php:1038
 Symfony\Component\Console\Application->doRunCommand() at (...)\vendor\symfony\framework-bundle\Console\Application.php:88
 Symfony\Bundle\FrameworkBundle\Console\Application->doRunCommand() at (...)\vendor\symfony\console\Application.php:312
 Symfony\Component\Console\Application->doRun() at (...)\vendor\symfony\framework-bundle\Console\Application.php:77
 Symfony\Bundle\FrameworkBundle\Console\Application->doRun() at (...)\vendor\symfony\console\Application.php:168
 Symfony\Component\Console\Application->run() at (...)\vendor\symfony\runtime\Runner\Symfony\ConsoleApplicationRunner.php:54
 Symfony\Component\Runtime\Runner\Symfony\ConsoleApplicationRunner->run() at (...)\vendor\autoload_runtime.php:29
 require_once() at (...)\bin\console:11

translation:extract [--prefix [PREFIX]] [--format [FORMAT]] [--dump-messages] [--force] [--clean] [--domain [DOMAIN]] [--sort [SORT]] [--as-tree [AS-TREE]] [--] <locale> [<bundle>]
```

**Comment by gndk** [comment]
Possibly related to #48422

**Comment by stof** [comment]
@Knallcharge do you have the PHP assertions enabled or no when getting this error ? If no, can you try enabling them and trying again, to see whether you get the error message from #48422 ?

**Comment by knallcharge** [comment]
@stof My phpinfo() says this: "assert.active => On => On", so I guess it's already on.

**Comment by seb-jean** [comment]
I have the same problem.

**Comment by voskobovich** [comment]
I have the same problem.

**Comment by DrakanAethril** [comment]
I have the same problem. If that may help, I have it when using this kind of notation : 

$translator->trans('Question of item %idQuizzQuestion% needs a text', ['%idQuizzQuestion%' => $i]);

When using trans without the additional parameter, that works fine.

**Comment by nicolas-grekas** [comment]
Should be fixed by #48451 indeed.
